src/rdb.c

```diff
@@ -3245,8 +3250,8 @@ int rdbLoadRioWithLoadingCtx(rio *rdb, int rdbflags, rdbSaveInfo *rsi, rdbLoadin
             continue;
         }
 
-        /* If there is no slot info, it means that it's either not cluster mode or we are trying to load legacy RDB
-         * file. In this case we want to estimate number of keys per slot and resize accordingly. */
+        /* If there is no slot info, it means that it's either not cluster mode or we are trying to load a legacy RDB
+         * file. In either cases we want to resize the db accordingly. */
         if (should_expand_db) {
```

Comment:
This comment is not correct. If we're in cluster mode and it's a legacy RDB file without slot info, we don't set `should_expand_db` so we don't resize the db.